crypto/threads_pthread.c

```diff
@@ -57,10 +57,14 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
      * We don't use recursive mutexes, but try to catch errors if we do.
      */
     pthread_mutexattr_init(&attr);
-#  if !defined(NDEBUG) && !defined(OPENSSL_NO_MUTEX_ERRORCHECK)
+#  if defined (__TANDEM) && defined (_SPT_MODEL_)
+    /* SPT does not define MUTEX attributes. */
+#  else
+#   if !defined(NDEBUG) && !defined(OPENSSL_NO_MUTEX_ERRORCHECK)
     pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK);
-# else
+#   else
```

Comment:
Yep, better alternative would be to do here: `# elif defined(PTHREAD_MUTEX_NORMAL)`

Reply:
@t8m Please have a look at my proposed revision.